cardinal/events/events_test.go

```diff
@@ -145,3 +148,40 @@ func TestEventsThroughSystems(t *testing.T) {
 	assert.Equal(t, counter1.Load(), int32(numberToTest*numberToTest))
 	assert.Equal(t, counter2.Load(), int32(numberToTest*numberToTest))
 }
+
+func TestEventHubLogger(t *testing.T) {
+	//replaces internal Logger with one that logs to the buf variable above.
+	var buf bytes.Buffer
+	bufLogger := zerolog.New(&buf)
+	cardinalLogger := ecslog.Logger{
+		&bufLogger,
+	}
+	w := ecs.NewTestWorld(t, ecs.WithLoggingEventHub(&cardinalLogger))
+	numberToTest := 5
+	for i := 0; i < numberToTest; i++ {
+		w.AddSystem(func(wCtx ecs.WorldContext) error {
+			wCtx.GetWorld().EmitEvent(&events.Event{Message: fmt.Sprintf("test")})
+			return nil
+		})
+	}
+	assert.NilError(t, w.LoadGameState())
+	ctx := context.Background()
+	waitForTicks := sync.WaitGroup{}
+	waitForTicks.Add(1)
+	go func() {
+		defer waitForTicks.Done()
+		for i := 0; i < numberToTest; i++ {
+			err := w.Tick(ctx)
+			assert.NilError(t, err)
+		}
+	}()
+	waitForTicks.Wait()
```

Comment:
any reason to spin up the go routine for the ticking? this has the same affect:

```go
	ctx := context.Background()
	for i := 0; i < numberToTest; i++ {
		assert.NilError(t, w.Tick(ctx))
	}
```

Reply:
It's left behind from the previous test where ticks need to run concurrently with clients and servers. Will remove.